src/eos-crash-metrics.c

```diff
@@ -0,0 +1,183 @@
+/* Copyright 2017 Endless Mobile, Inc. */
+
+#include <stdlib.h>
+#include <eosmetrics/eosmetrics.h>
+#include <ostree.h>
+
+#define PROGRAM_DUMPED_CORE_EVENT "ed57b607-4a56-47f1-b1e4-5dc3e74335ec"
+
+static GVariant *
+hashtable_to_variant (GHashTable *ht)
+{
+  g_autoptr(GList) keys = g_hash_table_get_keys (ht);
+  g_auto(GVariantBuilder) builder = G_VARIANT_BUILDER_INIT (G_VARIANT_TYPE ("a{sv}"));
+
+  g_variant_builder_init (&builder, G_VARIANT_TYPE ("a{sv}"));
+
+  GList *key_iter = keys;
+  for (; key_iter; key_iter = key_iter->next)
+    {
+      g_variant_builder_open (&builder, G_VARIANT_TYPE ("{sv}"));
+
+      g_variant_builder_add (&builder, "s", g_strdup (key_iter->data));
+      g_variant_builder_add (&builder, "v", g_variant_ref (g_hash_table_lookup (ht, key_iter->data)));
+
+      g_variant_builder_close (&builder);
+    }
+
+  return g_variant_builder_end (&builder);
+}
+
+static void
+report_crash (const char *binary,
+              gint16 signal,
+              gint64 timestamp,
+              const char *ostree_commit,
+              const char *ostree_url)
+{
+  g_autoptr(GHashTable) ht = g_hash_table_new_full (g_str_hash,
+                                                    g_str_equal,
+                                                    g_free,
+                                                    (GDestroyNotify) g_variant_unref);
+
+  g_hash_table_insert (ht, g_strdup ("binary"), g_variant_new_string (binary));
+  g_hash_table_insert (ht, g_strdup ("signal"), g_variant_new_int16 (signal));
+  g_hash_table_insert (ht, g_strdup ("timestamp"), g_variant_new_int16 (timestamp));
+  g_hash_table_insert (ht, g_strdup ("ostree_commit"), g_variant_new_string (ostree_commit != NULL ? ostree_commit : ""));
+  g_hash_table_insert (ht, g_strdup ("ostree_commit"), g_variant_new_string (ostree_url != NULL ? ostree_url : ""));
+
+  g_message("%s %s", ostree_commit, ostree_url);
```

Comment:
I think this could be removed. Otherwise, please make the message more descriptive, and add a space before paren